src/System.Text.Json/tests/Serialization/OptionsTests.cs

```diff
@@ -25,5 +49,69 @@ public static void DefaultBufferSize()
             options.DefaultBufferSize = 1;
             Assert.Equal(1, options.DefaultBufferSize);
         }
+
+        [Fact]
+        public static void AllowTrailingCommas()
+        {
+            Assert.Throws<JsonReaderException>(() => JsonSerializer.Parse<int[]>("[1,]"));
+
+            var options = new JsonSerializerOptions();
+            options.AllowTrailingCommas = true;
+
+            int[] value = JsonSerializer.Parse<int[]>("[1,]", options);
+            Assert.Equal(1, value[0]);
+        }
+
+        [Fact]
+        public static void WriteIndented()
+        {
+            var obj = new BasicCompany();
+            obj.Initialize();
+
+            // Verify default value.
+            string json = JsonSerializer.ToString(obj);
+            Assert.DoesNotContain("\r", json);
+
+            // Verify default value on options.
+            var options = new JsonSerializerOptions();
+            json = JsonSerializer.ToString(obj, options);
+            Assert.DoesNotContain("\r", json);
+
+            // Change the value on options.
+            options = new JsonSerializerOptions();
+            options.WriteIndented = true;
+            json = JsonSerializer.ToString(obj, options);
+            Assert.Contains("\r", json);
+        }
+
+        [Fact]
+        public static void ReadCommentHandling()
+        {
+            Assert.Throws<JsonReaderException>(() => JsonSerializer.Parse<object>("/* commment */"));
+
+            var options = new JsonSerializerOptions();
+
+            Assert.Throws<JsonReaderException>(() => JsonSerializer.Parse<object>("/* commment */", options));
+
+            options = new JsonSerializerOptions();
+            options.ReadCommentHandling = JsonCommentHandling.Allow;
+
+            JsonSerializer.Parse<object>("/* commment */", options);
+        }
+
+        [Fact]
+        public static void MaxDepthRead()
+        {
+            JsonSerializer.Parse<BasicCompany>(BasicCompany.s_data);
+
+            var options = new JsonSerializerOptions();
+
+            JsonSerializer.Parse<BasicCompany>(BasicCompany.s_data, options);
+
+            options = new JsonSerializerOptions();
+            options.MaxDepth = 1;
```

Comment:
Can you add a test for depths 63, 64, 65 as well (to catch potential off-by-once issues around default max depth)?

Reply:
Currently this is handled by the reader (throwing) so I just wanted to make sure MaxDepth here is applied to reader.MaxDepth.

However I will need to check ranges for https://github.com/dotnet/corefx/issues/36510 since I need to ensure the proper count is applied.